src/vulkan/transfer_buffer.cc

```diff
@@ -19,10 +19,20 @@
 namespace amber {
 namespace vulkan {
 
-TransferBuffer::TransferBuffer(Device* device, uint32_t size_in_bytes)
-    : Resource(device, size_in_bytes) {}
+TransferBuffer::TransferBuffer(Device* device,
+                               uint32_t size_in_bytes,
+                               Format* format)
+    : Resource(device, size_in_bytes) {
+  if (format)
+    format_ = device->GetVkFormat(*format);
+}
 
 TransferBuffer::~TransferBuffer() {
+  if (view_ != VK_NULL_HANDLE) {
+    device_->GetPtrs()->vkDestroyBufferView(device_->GetVkDevice(), view_,
```

Comment:
I don't think you need the `if()` check. From the spec:

```
If bufferView is not VK_NULL_HANDLE, bufferView must be a valid VkBufferView handle
```

which sounds like you can just pass in the VK_NULL_HANDLE

Reply:
Same applies to vkDestroyBuffer so I've removed the null check from there too.